test/services/schema-object-factory.spec.ts

```diff
@@ -42,21 +48,36 @@ describe('SchemaObjectFactory', () => {
     class Person {
       @ApiProperty({ enum: Role, enumName: 'Role' })
       role: Role;
+
       @ApiProperty({ enum: Role, enumName: 'Role', isArray: true })
       roles: Role[];
```

Comment:
A similar issue wasn't detected here with `roles` likely because `role` was processed first, where it isn't defined as an array.